src/utils/useBroadcast.js

```diff
@@ -0,0 +1,15 @@
+import { useEffect, useState, useRef } from 'react';
+
+export default function useBroadcast(broadcast, defaultValue) {
+  const [value, setValue] = useState(broadcast ? broadcast.getState() : defaultValue);
+  const unsubscribeRef = useRef();
+
+  useEffect(() => {
+    if (broadcast && !unsubscribeRef.current) {
+      unsubscribeRef.current = broadcast.subscribe(setValue);
+    }
+    return unsubscribeRef.current;
```

Comment:
I'm not sure we gain anything by using ref to hold onto the unsubscribe function. Could we do:

```ts
useEffect(() => {
  if (broadcast) {
    const unsubscribe = broadcast.subscribe(setValue);
    return unsubscribe;
  }
});
```

Reply:
I learn from you every day!

Reply:
Is it cool to sometimes not return a cleanup function?